lib/fpm/package/gem.rb

```diff
@@ -47,6 +47,22 @@ class FPM::Package::Gem < FPM::Package
 
   option "--version-bins", :flag, "Append the version to the bins", :default => false
 
+  option "--stagingdir", "STAGINGDIR",
+    "The directory where fpm installs the gem temporarily before conversion. " \
+    "Normally a random subdirectory of workdir."
+
+  # Override parent method
+  def staging_path(path=nil)
+    @gem_staging_path ||= attributes[:gem_stagingdir] || Stud::Temporary.directory("package-#{type}-staging")
```

Comment:
Well, yeah... but long-term this option won't be needed (once compilers obey https://reproducible-builds.org/specs/build-path-prefix-map/ ), so if we're going to have an intermediate workaround, let's make it simple, transparent, and scary so nobody uses it by accident.